lib/core/src/swapper/mod.rs

```diff
@@ -176,114 +424,172 @@ impl Swapper for BoltzSwapper {
         Ok(self.client.post_swap_req(&req)?)
     }
 
+    /// Get a chain pair information
+    fn get_chain_pairs(
+        &self,
+        payment_type: PaymentType,
+    ) -> Result<Option<ChainPair>, PaymentError> {
+        let pairs = self.client.get_chain_pairs()?;
+        let pair = match payment_type {
+            PaymentType::Receive => pairs.get_btc_to_lbtc_pair(),
+            PaymentType::Send => pairs.get_lbtc_to_btc_pair(),
+        };
+        Ok(pair)
+    }
+
     /// Get a submarine pair information
     fn get_submarine_pairs(&self) -> Result<Option<SubmarinePair>, PaymentError> {
         Ok(self.client.get_submarine_pairs()?.get_lbtc_to_btc_pair())
     }
 
+    /// Refund a cooperatively chain swap  
+    fn refund_chain_swap_cooperative(
+        &self,
+        swap: &ChainSwap,
+        output_address: &str,
+        broadcast_fees_sat: u64,
+    ) -> Result<String, PaymentError> {
+        let refund_keypair = swap.get_refund_keypair()?;
+        let swap_script = swap.get_lockup_swap_script()?;
+        match swap.payment_type {
+            PaymentType::Send => {
+                info!("Initiating cooperative refund for Chain Swap {}", &swap.id);
+                self.refund_swap_cooperative(
+                    swap.id.clone(),
+                    swap_script.as_liquid_script()?,
+                    &refund_keypair,
+                    output_address,
+                    broadcast_fees_sat,
+                )
+            }
+            PaymentType::Receive => Err(PaymentError::Generic {
+                err: format!("Cannot refund a receive Chain Swap {}", swap.id),
```

Comment:
We actually do need to refund on the receive case. The user sends btc to the script address and if boltz doesn't claim them then we need to refund the user to an address he/she is passing.
I suggested this kind of refund will be in a different API but we still need that logic on the swapper.
Regardless I think worth considering call this method refund_send_chain_swap_cooperative and let it handle only the send case.